app/models/policy.rb

```diff
@@ -0,0 +1,108 @@
+class Policy < ActiveRecord::Base
+  validates :content_id, presence: true, uniqueness: true
+  validates :name, :slug, presence: true, uniqueness: true
+  validates :description, presence: true
+
+  validate :applicable_to_at_least_one_nation
+  validate :alternative_urls_are_valid
+
+  has_many :policy_relations
+  has_many :related_policies, class_name: 'Policy', through: :policy_relations, source: :related_policy
+
+  has_many :inverse_policy_relations, class_name: 'PolicyRelation', foreign_key: 'related_policy_id'
+  has_many :parent_policies,  through: :inverse_policy_relations, source: :policy
+
+  before_validation on: :create do |object|
+    object.slug = object.name.to_s.parameterize
+    object.content_id = SecureRandom.uuid
+  end
+
+  scope :areas, -> { joins("LEFT OUTER JOIN policy_relations
+                            ON policies.id = policy_relations.related_policy_id
+                            WHERE policy_relations.related_policy_id IS NULL") }
+
+  after_save :publish!
+  after_save :republish_parents!
+
+  # Virtual attribute used to identify a new record as a programme
+  attr_writer :programme
+
+  def publish!
+    publish_content_item!
+    add_to_search_index!
+  end
+
+  def republish!
+    publish_content_item!('minor')
+    add_to_search_index!
+  end
+
+
+  def base_path
+    "/government/policies/#{slug}"
+  end
+
+  def possible_nations
+    [
+      :england,
+      :northern_ireland,
+      :scotland,
+      :wales,
+    ]
+  end
+
+  def applicable_nations
+    applicable_nations = possible_nations.select { |n|
+      self.send(n) == true
+    }
+  end
+
+  def inapplicable_nations
+    possible_nations - applicable_nations
+  end
+
+  def programme
+    @programme || parent_policies.any?
```

Comment:
That's an interesting question. To be honest this is the one part of the merge that feels like a temporary hack. The main thing I wanted to avoid was baking in unnecessary assumptions about the differences between programmes and areas into the data model, hence the use of a virtual attribute.  As the differences between programmes and areas develops and the functionality of the policy publisher is extended, I expect that this code to evolve/be replaced. In my mind, adding the restriction now would mainly make the code harder to change because it would codify the restriction, when we technically don't need it, at least not yet.